dataactbroker/handlers/fileHandler.py

```diff
@@ -639,6 +639,123 @@ def upload_detached_file(self, create_credentials):
+
+            # get the cgac code associated with this sub tier agency
+            job_data["cgac_code"] = cgac_code
+            job_data["d2_submission"] = True
+
+            # convert submission start/end dates from the request into Python date objects
+            try:
+                job_data['reporting_start_date'] = formatted_start_date
+                job_data['reporting_end_date'] = formatted_end_date
+
+            except ValueError:
+                raise ResponseException("Date must be provided as DD/MM/YYYY", StatusCode.CLIENT_ERROR, ValueError)
+
+            # the front-end is doing date checks, but we'll also do a few server side to ensure everything is correct
+            # when clients call the API directly
+            if job_data.get('reporting_start_date') > job_data.get('reporting_end_date'):
+                raise ResponseException("Submission start date {} is after the end date {}".format(
+                        job_data.get('reporting_start_date'), job_data.get('reporting_end_date')),
+                        StatusCode.CLIENT_ERROR)
+
+            if not current_user_can('writer', job_data["cgac_code"]):
+                raise ResponseException("User does not have permission to create jobs for this agency",
+                                        StatusCode.PERMISSION_DENIED)
+
+            submission = create_submission(g.user.user_id, job_data, existing_submission_obj)
+            sess.add(submission)
+            sess.commit()
+            if existing_submission_obj is None:
+                sub_tier_agency_id = sub_tier_agency.sub_tier_agency_id
+                sub_tier_affiliation = SubmissionSubTierAffiliation(submission_id=submission.submission_id,
+                                                                    sub_tier_agency_id=sub_tier_agency_id)
+                sess.add(sub_tier_affiliation)
+                sess.commit()
+
+                # build fileNameMap to be used in creating jobs
+            self.build_file_map(request_params, ['detached_award'], response_dict, upload_files)
+
+            self.create_response_dict_for_submission(upload_files, submission, existing_submission,
+                                                     response_dict, create_credentials)
+            return JsonResponse.create(StatusCode.OK, response_dict)
+        except (ValueError, TypeError, NotImplementedError) as e:
+            return JsonResponse.error(e, StatusCode.CLIENT_ERROR)
+        except ResponseException as e:
+            # call error route directly, status code depends on exception
+            return JsonResponse.error(e, e.status)
+        except Exception as e:
+            # unexpected exception, this is a 500 server error
+            return JsonResponse.error(e, StatusCode.INTERNAL_ERROR)
+        except:
+            return JsonResponse.error(Exception("Failed to catch exception"), StatusCode.INTERNAL_ERROR)
+
+    def update_detached_file(self, create_credentials):
```

Comment:
Why are we created an entire new function that does 99% of the same stuff? `upload_detached_files` has a check for existing submissions. You'll have to update how it works slightly but in general you can just use the existing code. If you look at the regular upload, I believe the code calls the same function whether it's an existing submission or a new one.